cpp/include/cudf/join.hpp

```diff
@@ -383,11 +383,11 @@ class hash_join {
    * constructed with null check.
    *
    * @return A pair of columns [`left_indices`, `right_indices`] that can be used to construct
-   * the result of performing a full join between two tables with `build` and `probe`
+   * the result of performing a full join between two tables w[[nodiscard]] ith `build` and `probe`
```

Comment:
Seems like a typo.
```suggestion
   * the result of performing a full join between two tables with `build` and `probe`
```

Reply:
Ugh yeah this happened a few times. I thought I had caught all of these, but I guess not. I'll do another pass, thanks.